src/Extents.jl

```diff
@@ -60,4 +80,89 @@ function extent end
 extent(extent) = nothing
 extent(extent::Extent) = extent
 
+"""
+    intersects(ext1::Extent, ext2::Extent)
+
+Check if two `Extent` objects intersect.
+
+Returns `true` if the extents of all common dimensions share some values
+including just the edge values of thier range.
+
+Dimensions that are not shared are ignored. The order of dimensions is also ignored.
```

Comment:
Might be good to have a `;strict=false`, that when enabled, requires shared dimensions?